url913d8566d1_AnsBalin_rbn_tgz-pJ8-MainJ8/src/Main.java

```diff
@@ -0,0 +1,1121 @@
+    return netResult;
+    
+  }
+  
+  public String reactionToString(Reaction R){
+    
+    String str = "";
+    
+    for(int i=0; i<R.getReactants().length; i++){
+      str = str+R.getMolReactants().get(i).toStringf()+" ";
+      if(i!=R.getReactants().length-1){
+        str = str+"+ ";
+      }
+    }
+    str=str+"---> ";
+    for(int i=0; i<R.getProducts().length; i++){
+      str = str+R.getMolProducts().get(i).toStringf()+"  ";
+      if(i!=R.getProducts().length-1){
+        str = str+"+ ";
+      }
+    }
+    return str;
+  
+  }
+  
+  public void formBond(Molecule A, Molecule B, Molecule Asub, Molecule Bsub, int bondingSite1, int bondingSite2){
+    
+    
+    
+    A.calculateMoleculeID(0,0,new int[0]);
+    B.calculateMoleculeID(0,0, new int[0]);
+    ArrayList<Molecule> molsA = A.split(Asub.getMolID());
+    A.unFlagAll();
+    ArrayList<Molecule> molsB = B.split(Bsub.getMolID());
+    B.unFlagAll();
+    if(molsA.size()>0){
+      //System.out.printf("!!!!!!!!!!!! %d\n", molsA.size());
+      
+    }
+    //System.out.printf(""+Arrays.toString(Asub.getMolID())+"\n");
+    
+    ArrayList<Network> after = new ArrayList<Network>();
+    
+    after.addAll( moleculeArrToNetworkArr(molsA) );
+    after.addAll( moleculeArrToNetworkArr(molsB) );
+    after.add( bond(Asub, Bsub, bondingSite1, bondingSite2) );
+
+    
+    numReactions++;
+    ArrayList<Molecule> reactants = new ArrayList<Molecule>();
+    ArrayList<Molecule> products = new ArrayList<Molecule>();
+    
+    reactants.add(A);
+    reactants.add(B);
+    //
+    //System.out.printf("\n\t"+A.toStringf()+" + "+B.toStringf()+" --> ");
+    bucket.remove(A);
+    bucket.remove(B);
+    library.get(A.getID()).decrPop();
+    library.get(B.getID()).decrPop();
+    
+    
+    
+    for(int i=0; i<molsA.size(); i++){
+      
+      bucket.add(molsA.get(i));
+      products.add(molsA.get(i));
+      library.get(molsA.get(i).getID()).incrPop();
+     // System.out.printf(""+molsA.get(i).toStringf()+" + ");
+    }
+    
+    for(int i=0; i<molsB.size(); i++){
+        
+        bucket.add(molsB.get(i));
+        products.add(molsB.get(i));
+        library.get(molsB.get(i).getID()).incrPop();
+       // System.out.printf(""+molsB.get(i).toStringf()+" + ");
+      }
+    
+    
+    
+    int index = libraryUpdate(after.get(after.size()-1), Asub, Bsub);
+    bucket.add(new Molecule(index, Asub, Bsub));
+    library.get(index).incrPop();
+    bucket.get(bucket.size()-1).setSize(Asub.getSize()+Bsub.getSize());
+    
+    products.add(bucket.get(bucket.size()-1));
+    
+    checkCatalysis(reactants, products);
+    
+    Reaction R = new Reaction(reactants, products);
+    int reactionIndex = reactionsUpdate(R);
+    reactions.get(reactionIndex).incrCount();
+    
+
+    //System.out.printf(""+bucket.get(bucket.size()-1).toStringf()+"\n");
+
+    
+  }
+  
+  public void checkCatalysis( ArrayList<Molecule> reactants, ArrayList<Molecule> products ){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.